packages/core/src/controller/Controller.ts

```diff
@@ -32,8 +32,8 @@ export default class Controller {
       entities: {},
       results: {},
     },
-  }: ConstructorProps) {
-    this.dispatch = dispatch;
+  }: ConstructorProps = {}) {
+    this.dispatch = dispatch ?? (() => Promise.reject('dispatch not set'));
```

Comment:
```suggestion
    this.dispatch = dispatch ?? () => Promise.reject('dispatch not set');
```
altho it seems like you could do `dispatch = () => Promise.reject('dispatch not set')` in the constructor args, unless you care about supporting `null`

Reply:
is this just removing the parens? This was required by the lint parser

Reply:
yes, and that's very weird but fine i guess.

the real comment was about using defaulting vs `??`, unless the type allows for `null`

Reply:
?? works on undefined right? the parameter is optional

Reply:
sure, but it also works on null.

if you don't care about null, it seems a bit cleaner to do it declaratively in the constructor params as a default instead of imperatively in the constructor body with `??`.

Reply:
Hmm, in that case you're constructing the function every call no matter what. That's probably pretty minor, but I also don't see a huge gain from using default.

Reply:
i'm pretty sure it's only evaluated when the default is triggered:
```js
function f(x = console.log('*')) {} [f(), f(null), f({})]
```
only logs once.

Reply:
ahh, that actually makes sense now that i think about it.